decksite/sql/46.sql

```diff
@@ -0,0 +1,8 @@
+CREATE TABLE IF NOT EXISTS confidence (
```

Comment:
Alternative names for this table – deck_confidence or _deck_confidence.

There's a little stylistic confusion about what we call tables and it's my fault.

I think we should use the following naming scheme:

foo – In the singular, a table that contains a row per foo (person, deck).
foo_bar – The conjunction of a foo and a bar and associated data (deck_match, deck_card).
_foo_stats, _baz – Tables that are deleted and recreated in their entirety from values elsewhere during preaggregation. No original data lives in these tables.

Which leaves us with just `deck_cache` and `confidence` as things that don't fit into any of these buckets. deck_cache is "things that are derivable from other things in the db + code logic but we tear it down and rebuild it deck-by-deck during reprime_cache and omw gets updated separately" and confidence is remarkably similar to that. There's no underivable data in confidence you just apply code logic to the existing database and out comes confidence. That makes it more like the _ tables with some differences in how it is formed than like the actual data tables.

Options:

1. Rename deck_cache to _deck_cache. Call confidence _deck_confidence. Ignore how these tables are formed and just use the _ to indicate "if you deleted this table and there were no backups we wouldn't care".
2. Leave deck_cache as is and rename confidence to confidence_cache.
3. Leave deck_cache as is and rename confidence to deck_confidence_cache.

Separately thinking about all this leads me to the question:

Why not put the confidence column into deck_cache?

It is a derivable thing about a deck generated by a job that can change going forward just like omw or wins or the sanitized name. It seems like a pretty perfect fit now that I've thought about it.

None of this changes the shippabilty of this code but perhaps we want to push only the "final" version if we're convinced confidence belongs in deck_cache?

(Also, does archetype_id belong in deck_cache??? I should probably close this can of worms now.)